EnhancementofUnderwaterImages/GuidedFilter.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import cv2
+
+class GuidedFilter:
+    
+    # def __init__(self, I, radius=5, epsilon=0.4):
+    def __init__(self, I, radius, epsilon):
+
+        self._radius = 2 * radius + 1
+        self._epsilon = epsilon
+        self._I = self._toFloatImg(I)
+        self._initFilter()
+
+        # print('radius',self._radius)
+        # print('epsilon',self._epsilon)
+
+    def _toFloatImg(self, img):
+        if img.dtype == np.float32:
+            return img
+        return ( 1.0 / 255.0 ) * np.float32(img)
+
+    def _initFilter(self):
+        I = self._I
+        r = self._radius
+        eps = self._epsilon
+
+        Ir, Ig, Ib = I[:, :, 0], I[:, :, 1], I[:, :, 2]
+
+        # self._Ir_mean = cv2.blur(Ir, (r, r))
```

Comment:
explain the reason for commenting these parts and when we might uncomment and use these. Else, remove these lines